gear-programs/checkpoint-light-client/src/wasm/state.rs

```diff
@@ -22,163 +22,47 @@ pub struct ReplayBackState {
 }
 
 #[derive(Debug, Clone)]
-pub struct Checkpoints<const N: usize> {
-    checkpoints: Box<CircularBuffer<N, Hash256>>,
-    slots: Vec<(usize, Slot)>,
-}
+pub struct Checkpoints<const N: usize>(Box<CircularBuffer<N, (Slot, Hash256)>>);
```

Comment:
I suggest introducing here struct for `(Slot, Hash256)` as it's not clear what are these